src/h5web/providers/models.ts

```diff
@@ -32,17 +31,17 @@ export interface Dataset<S extends Shape = Shape, T extends DType = DType>
   rawType?: unknown;
 }
 
-export type NumArrayDataset = Dataset<ArrayShape, NumericType>;
-
 export interface Datatype<T = DType> extends Entity {
   kind: EntityKind.Datatype;
   type: T;
   rawType?: unknown;
 }
 
-export interface Link<T extends HDF5Link = HDF5Link> extends Entity {
-  kind: EntityKind.Link;
-  rawLink: T;
+export type LinkClass = 'Hard' | 'Soft' | 'External';
+export interface Link {
+  class: LinkClass;
+  path?: string;
+  file?: string;
```

Comment:
As discussed, those are the only information that seem relevant for H5Web. A hard link is represented with just `class`; a soft link with `class` and `path`; and an external link with `class`, `path` and `file`.

Links are only really used by the metadata viewer to display the path/file if present, so we don't need any more typing than that.